### ISSUE TITLE
The implementation of loss function

### ISSUE THREAD

**Issue by YuxinZhaozyx** [abhay97ps/Anomaly-Detection-in-Surveillance-Videos#14]
The loss function in your paper is 

![image](https://user-images.githubusercontent.com/32972764/54752790-427f2400-4c1a-11e9-9113-b06bcef6e89a.png)

![image](https://user-images.githubusercontent.com/32972764/54752765-2d09fa00-4c1a-11e9-8248-79688a928c9d.png)

However, you implement it in the following way.

https://github.com/abhay97ps/Anomaly-Detection-in-Surveillance-Videos/blob/4be07aba4a8dda7cea882f33782595622d248d3e/Anomaly_NN.py#L51

https://github.com/abhay97ps/Anomaly-Detection-in-Surveillance-Videos/blob/4be07aba4a8dda7cea882f33782595622d248d3e/Anomaly_NN.py#L66-L77

The ![image](https://user-images.githubusercontent.com/32972764/54752893-95f17200-4c1a-11e9-81ae-4b16bcdd8503.png) is implemented as a placeholder `extra` and calculated using python float type. Won't it make it impossible to back propagate gradient to `extra`? And the optimizer only minimizes the  ![image](https://user-images.githubusercontent.com/32972764/54753611-6e030e00-4c1c-11e9-93b0-fd6eec913296.png) .

Shouldn't the ![image](https://user-images.githubusercontent.com/32972764/54752893-95f17200-4c1a-11e9-81ae-4b16bcdd8503.png) be implemented in tensorflow functions rather than pure python?

**Comment by fzp8656342** [comment]
It seems the weight didn't get update every iteration in here. Have you figure out how to do it correctly?
